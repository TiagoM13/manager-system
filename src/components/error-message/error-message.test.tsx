import React from 'react';

import { render } from '@testing-library/react';

import { ErrorMessage } from '.';

describe('<ErrorMessage/>', () => {
  test('should render correctly component', () => {
    const message = 'Error Message';
    const { getByText, asFragment } = render(<ErrorMessage error={message} />);

    expect(getByText(message)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('should not render anything when error prop is not provided', () => {
    const { container } = render(<ErrorMessage />);

    expect(container.firstChild).toBeNull();
  });
});
