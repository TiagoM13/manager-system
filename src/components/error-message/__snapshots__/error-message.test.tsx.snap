// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage/> should render correctly component 1`] = `
<DocumentFragment>
  <span
    class="block text-sm text-red-500 mt-1"
  >
    Error Message
  </span>
</DocumentFragment>
`;
